UPGRADE-2.0.md

```diff
@@ -1,5 +1,126 @@
 # UPGRADE FROM `1.14` TO `2.0`
 
+## Initial part
+
+Even if your app is barely customized, it will require some manual adjustments before it can run again. Depending on
+whether you use Symfony Flex, some of these changes may be applied automatically, but it’s important to check them
+manually regardless.
+
+* Packages configuration changes:
+
+```md
+#config/packages/_sylius.yaml
+
+imports:
+...
++   - { resource: "@SyliusPayumBundle/Resources/config/app/config.yaml" }
+
+...
+
+sylius_payment:
+    resources:
++       gateway_config:
++           classes:
++               model: App\Entity\Payment\GatewayConfig
+
+...
+
+sylius_payum:
+    resources:
+-       gateway_config:
+-           classes:
+-               model: App\Entity\Payment\GatewayConfig
+```
+
+* API firewalls have been renamed and user checkers have been configured on firewalls
+  with `security.user_checker.chain.<firewall>` service:
+
+```diff
+#config/packages/security.yaml
+
+security:
+    firewalls:
+        admin:
+            ...
++           user_checker: security.user_checker.chain.admin
+-       new_api_admin_user:
++       api_admin:
+            ...
++           user_checker: security.user_checker.chain.api_admin
+-       new_api_shop_user:
++       api_shop:
+            ...
++           user_checker: security.user_checker.chain.api_shop
+        shop:
+            ...
++           user_checker: security.user_checker.chain.shop
+```
+
+To reflect these changes, the route configuration must be updated accordingly:
+
+```diff
+# config/routes/sylius_api.yaml
+
+sylius_api:
+    resource: "@SyliusApiBundle/Resources/config/routing.yml"
+-   prefix: "%sylius.security.new_api_route%"
++   prefix: "%sylius.security.api_route%"
+
+```
```

Comment:
This is not related to the security configuration above, it's a change of parameters and fits more under `routing changes`